datastream-server/src/main/java/com/linkedin/datastream/server/providers/FileBasedPartitionThroughputProvider.java

```diff
@@ -0,0 +1,120 @@
+/**
+ *  Copyright 2021 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.server.providers;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Iterator;
+
+import org.apache.commons.lang3.StringUtils;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.type.TypeReference;
+
+import com.linkedin.datastream.server.ClusterThroughputInfo;
+import com.linkedin.datastream.server.PartitionThroughputInfo;
+
+
+/**
+ * An implementation of {@link PartitionThroughputProvider} which reads throughput data from a resource file.
+ * Used for testing purposes.
+ */
+public class FileBasedPartitionThroughputProvider implements PartitionThroughputProvider {
+  private static final String RESOURCE_FILE_NAME = "partitionThroughput.json";
```

Comment:
Will it make more sense to get this via Constructor, so that there is a knob to test additional json files with partition input without tweaking this file

Reply:
Done